src/sentry/workflow_engine/models/detector.py

```diff
@@ -255,37 +269,45 @@ def evaluate_group_key_value(
 
         is_active = status != DetectorPriorityLevel.OK
 
+        # TODO: We'll increment and change these later, but for now they don't change so just pass an empty dict
+        self.enqueue_counter_update(group_key, {})
+
         if state_data.active != is_active or state_data.status != status:
+            self.enqueue_state_update(group_key, is_active, status)
             return DetectorEvaluationResult(
+                group_key,
                 is_active,
                 status,
                 {},
-                # TODO: We actually always need to be able to commit this. I'll move this out of the result
-                DetectorStateData(
-                    group_key=group_key,
-                    active=is_active,
-                    status=status,
-                    dedupe_value=dedupe_value,
-                    # TODO: We'll increment and change these later, but for now they don't change so just pass an empty dict
-                    # so we no-op
-                    counter_updates={},
-                ),
             )
         return None
 
-    def build_dedupe_value_key(self, group_key: str | None) -> str:
+    def enqueue_dedupe_update(self, group_key: DetectorGroupKey, dedupe_value: int):
```

Comment:
If it was enqueueing multiple updates, what would it do? I think the last write would win, since we're updating the same key. So this actually has the same behaviour as-is. I'm not sure that we'd want to throw an error instead